File: Python/Java/3355ZeroArrayTransformationI.py
class Solution(object):
    def isZeroArray(self, nums, queries):
        """
        :type nums: List[int]
        :type queries: List[List[int]]
        :rtype: bool
        """

        line = [0] * (len(nums) + 1)
        decr = 0

        for left, right in queries:
            line[left] += 1
            line[right + 1] -= 1

        for i, num in enumerate(nums):
            decr += line[i]
            if (decr < num):
                return False

        return True
        

File: Java/3355ZeroArrayTransformationI.java
class Solution {
    public boolean isZeroArray(int[] nums, int[][] queries) {
        int[] line = new int[nums.length + 1];
        int decr = 0;

        for (int[] query : queries) {
            final int left = query[0];
            final int right = query[1];

            ++line[left];
            --line[right + 1];
        }

        for (int i = 0; i < nums.length; ++i) {
            decr += line[i];
            if (decr < nums[i])
                return false;
        }

        return true;
    }
}
